webui/react/src/utils/types.ts

```diff
@@ -234,3 +235,22 @@ export const tsbMatchesSource = (tensorboard: Command, source: TBSource): boolea
       return false;
   }
 };
+
+export const getMetricValue = (workload?: Workload, metricName?: string): number | undefined => {
+  const metricsWl = workload as MetricsWorkload;
+  if (!workload || !isMetricsWorkload(metricsWl) || !metricsWl.metrics) return undefined;
+
+  metricName = metricName || Object.keys(metricsWl.metrics)[0];
+
+  return metricsWl.metrics[metricName];
+};
+
+export const getBatchNumber = (
+  data: {batch: number} | {numBatches: number, priorBatchesProcessed: number},
+): number => {
+  if ('batch' in data) {
+    return data.batch;
+  } else {
+    return data.numBatches + data.priorBatchesProcessed;
+  }
+};
```

Comment:
**suggestion:** taking a quick look at `utils` folder, the lines are kind of blurred between `step.ts`, `trial.ts` and these functions. What if we grouped them into just a bucket of `experiment.ts`?

Reply:
hmm maybe rename steps into experiments? I think there is more we could do here all of which would add a lot of changes not directly related to this PR leave for a separate PR if we think it warrants it?